src/backends/torch/optim/ranger.cc

```diff
@@ -174,6 +179,8 @@ namespace dd
                 grad.add_(-grad.mean(torch::IntArrayRef(dim), true));
               }
 
+            grad.clamp_(-options.clip(), options.clip());
```

Comment:
Just checking: this is outside the `rectified` conditional.

Reply:
clipping is now explicitly controllable (and forced to true in case of rectified)